README.md

```diff
@@ -31,43 +33,62 @@ When connecting to a PostgreSQL database running on a Windows host from a WSL2 i
+
+[FastAPI](https://fastapi.tiangolo.com/) is used to support the REST APIs. Non-blocking I/O is achieved using [asyncio](https://docs.python.org/3/library/asyncio.html) and this influences the PostgreSQL. MongoDB and Redis clients used.
+
+[Pydantic](https://docs.pydantic.dev/) is used to model externally visible schemas. The generated OpenAPI documentation can be found [here](http://localhost:5510/docs). There is also a [redoc](http://localhost:5510/redoc) version.
+
+Other than that the reliance on external code is kept to a minimum so that the project is easy for someone new to understand what's going on.
 
 ## Project Structure
 
-The project uses the [NestJS](https://docs.nestjs.com/) framework with [Mikro-ORM](https://mikro-orm.io/docs/installation).
+This is mainly an SQL-based project so the Python application framework is kept to a minimum. The main folders (inside the `query` folder) are as follows:
+
+ - Root folder: API routes (in `main.py`), schedule definition and other core dependencies, i.e. PostgreSQL, MongoDB and Redis
+ - `tables`: Each table, or set of very similar tables (like tags) has a module. This contains all of the SQL for modifying the data and structure of the table, so even `migrations` call out to functions in these modules. The `tables` modules will contain limited business logic, mostly where this would be the equivalent of a database trigger
+ - `migrations`: These manage database schema updates. Note in most cases the SQL itself is in the relevant `tables` module. The migration history is stored in a table called `mikro_orm_migrations` for backward compatibility with the previous NestJS / Mikro-ORM implementation of this project
+ - `services`: This is where most of the complex business logic lives, where interactions between multiple tables are coordinated
+ - `models`: These are the [Pydantic](https://docs.pydantic.dev/) classes used to validate API requests and generate OpenAPI documentation.
+ - `views`: The SQL definition for any views that are created in the database during migrations
+ - `assets`: Non-python resources
 
-The entrypoint is main.js which runs database migrations and starts the service.
+The entrypoint is main.py which runs database migrations and starts the service and scheduler.
 
-The main business logic is in the domain/services folder and the controllers route through to here. Extensive use is made of the Mikro-ORM [EntityManager](https://mikro-orm.io/docs/entity-manager) here to interact with the database, although in some cases more raw SQL is used to optimise performance.
+## Design Philosophy
 
-The domain/entities folder defines the main entities and is used to automatically generate migrations using the [migration:create](https://mikro-orm.io/docs/migrations#initial-migration) npm task.
+Most of the logic in this project is in the actual SQL used to update and query data. There has therefore been conscious decision to avoid using any kind of ORM to hade the SQL from the code.
```

Comment:
```suggestion
Most of the logic in this project is in the actual SQL used to update and query data. There has therefore been conscious decision to avoid using any kind of ORM to hide the SQL from the code.
```